test/end-to-end/test/test_editRecipe.js

```diff
@@ -120,14 +126,19 @@ describe('Edit Recipe Page', function () {
         const nightmare = new Nightmare();
         nightmare
           .goto(editRecipePage.url)
+          .wait(editRecipePage.componentListItemRootElement)
           .wait(editRecipePage.btnCreateCompos)
           .click(editRecipePage.btnCreateCompos)
-          .wait(createComposPage.btnCreate)
+          .wait(page => document.querySelector(page.dialogRootElement).style.display === 'block'
+            , createComposPage)
```

Comment:
The Create Composition dialog page is already in the Dom even the dialog does not pop up. The way to control this page visible is update the div property style from "display: none;" to "display: block;". So the whether Create button in the Dom tree or not cannot prove the page is ready to use.